src/rts/GameState.java

```diff
@@ -344,24 +344,32 @@ public boolean issueSafe(PlayerAction pa) {
             }
             
             // get the unit that corresponds to that action (since the state might have been cloned):
-            if (pgs.units.indexOf(p.m_a)==-1) {
```

Comment:
in the original code, this `indexOf()` call internally also did its own loop through all units. With my new code, there's only a single loop in total through all units, instead of two.